buildroot/share/PlatformIO/ldscripts/anet_et4.ld

```diff
@@ -0,0 +1,184 @@
+/*
+*****************************************************************************
+**
+**  File        : LinkerScript.ld
```

Comment:
This looks like a duplicate file which should be deleted.

Reply:
It is already in the variant folder, which is where it should be. It seems like mostly STM32F1 boards that have scripts here.

Most of the other boards in this HAL have them only in the variant folder.

Reply:
This isn't the same as [Marlin/buildroot/share/PlatformIO/variants/ANET_ET4/ldscript.ld](https://github.com/MarlinFirmware/Marlin/blob/4986377dd3d53b4990b170bc1e0f112561ec3ebb/buildroot/share/PlatformIO/variants/ANET_ET4/ldscript.ld) and removing it causes the build to fail.

Reply:
Or the two ldscript are equal, or one of them is wrong. Keeping both will lead to confusion. The script `buildroot/share/PlatformIO/scripts/anet_et4.py` is forcing to use this file.
But I think we should "join" this in favor of the variant version (even if we need to fix the variant file).

I think we should try to use our generic scripts too, instead of specific ones, for the new STM32 HAL. So we avoid having one script per board...